search-commons/src/test/java/no/unit/nva/search/IndexDocumentTest.java

```diff
@@ -70,21 +85,59 @@ public void toIndexDocumentReturnsIndexDocumentWitEmptyContributorListWhenPublic
     @Test
     public void toIndexDocumentReturnsIndexDocumentWithNoDateWithPublicationDateIsNull()
         throws MalformedURLException, InvalidIssnException {
-        Publication publication = PublicationGenerator.publicationWithIdentifier();
+        Publication publication = publicationWithIdentifier();
         publication.getEntityDescription().setDate(null);
         IndexDocument indexDocument = IndexDocument.fromPublication(publication);
         assertThat(indexDocument.getPublicationDate(),is(nullValue()));
     }
 
 
+    @Test
+    void toJsonStringSerializesUrisAsIdentifiers()
+            throws InvalidIsbnException, InvalidUnconfirmedSeriesException {
+
+        final URI bookSeriesId = randomPublicationChannelsUri();
+        final URI publisherId = randomPublicationChannelsUri();
+        final PublishingHouse publisher = publishingHouseWithUri(publisherId);
+        EntityDescription entityDescription = createSampleEntityDescriptionBook(bookSeriesId, publisher);
+        Publication publication = createPublicationWithEntityDescription(entityDescription);
+        assertThat(publication, doesNotHaveEmptyValuesIgnoringFields(IGNORED_PUBLICATION_FIELDS));
```

Comment:
Do not copy paste the code in tests.  Suggestion:
```
    @Test
    void toJsonStringSerializesUrisAsIdentifiers()
        throws InvalidIsbnException {

        Publication publication = createSampleBookInABookSeriesFromAPublisher(publishingHouseWithUri());

        assertThat(publication, doesNotHaveEmptyValuesIgnoringFields(IGNORED_PUBLICATION_FIELDS));

        IndexDocument actualDocument = IndexDocument.fromPublication(publication);
        assertThat(actualDocument, doesNotHaveEmptyValuesIgnoringFields(IGNORED_INDEXED_DOCUMENT_FIELDS));
    }

    private Publication createSampleBookInABookSeriesFromAPublisher(PublishingHouse publishingHouse)
        throws InvalidIsbnException {
        EntityDescription entityDescription = createSampleEntityDescriptionBook(publishingHouse);
        return createPublicationWithEntityDescription(entityDescription);
    }

    @Test
    public void toJsonPreservesRegistredDataWhenIdentifierIsUriFromPublicationChannels() throws Exception {

        Publication publication = createSampleBookInABookSeriesFromAPublisher(publishingHouseWithUri());

        IndexDocument actualDocument = IndexDocument.fromPublication(publication);
        assertThat(actualDocument, doesNotHaveEmptyValuesIgnoringFields(IGNORED_INDEXED_DOCUMENT_FIELDS));
        assertEquals(actualDocument.getReference(), publication.getEntityDescription().getReference());
    }

    @Test
    public void toJsonPreservesRegistredDataDataWhenPublisherIsUnconfirmed() throws Exception {

        Publication publication = createSampleBookInABookSeriesFromAPublisher(unconfirmedPublishingHouse());

        IndexDocument actualDocument = IndexDocument.fromPublication(publication);
        assertThat(actualDocument, doesNotHaveEmptyValuesIgnoringFields(IGNORED_INDEXED_DOCUMENT_FIELDS));
        assertEquals(actualDocument.getReference(), publication.getEntityDescription().getReference());
    }

//These methods belong in PublicationGenerator
   public static PublishingHouse publishingHouseWithUri() {
        return new Publisher(randomPublicationChannelsUri());
    }

    public static PublishingHouse unconfirmedPublishingHouse() {
        return new UnconfirmedPublisher(randomString());
    }


```

Reply:
Tests changed without duplicates